microprofile/health-smallrye/src/main/java/org/wildfly/extension/microprofile/health/HealthReporterService.java

```diff
@@ -40,30 +51,52 @@
 public class HealthReporterService implements Service<HealthReporter> {
 
     private static HealthReporter healthReporter;
+    private final Supplier<ModelControllerClientFactory> modelControllerClientFactory;
+    private final Supplier<Executor> managementExecutor;
     private String emptyLivenessChecksStatus;
     private String emptyReadinessChecksStatus;
+    private LocalModelControllerClient modelControllerClient;
 
     static void install(OperationContext context, String emptyLivenessChecksStatus, String emptyReadinessChecksStatus) {
-        context.getCapabilityServiceTarget()
-                .addCapability(RuntimeCapability.Builder.of(HEALTH_REPORTER_CAPABILITY, SmallRyeHealthReporter.class).build())
-                .setInstance(new HealthReporterService(emptyLivenessChecksStatus, emptyReadinessChecksStatus))
+
+        CapabilityServiceBuilder<?> serviceBuilder = context.getCapabilityServiceTarget()
+                .addCapability(RuntimeCapability.Builder.of(HEALTH_REPORTER_CAPABILITY, SmallRyeHealthReporter.class).build());
+
+        Supplier<ModelControllerClientFactory> modelControllerClientFactory = serviceBuilder.requires(context.getCapabilityServiceName(CLIENT_FACTORY_CAPABILITY, ModelControllerClientFactory.class));
+        Supplier<Executor> managementExecutor = serviceBuilder.requires(context.getCapabilityServiceName(MANAGEMENT_EXECUTOR, Executor.class));
+
+        serviceBuilder.setInstance(new HealthReporterService(modelControllerClientFactory, managementExecutor, emptyLivenessChecksStatus, emptyReadinessChecksStatus))
                 .install();
     }
 
-    private HealthReporterService(String emptyLivenessChecksStatus, String emptyReadinessChecksStatus) {
+    private HealthReporterService(Supplier<ModelControllerClientFactory> modelControllerClientFactory,
+                                  Supplier<Executor> managementExecutor,
+                                  String emptyLivenessChecksStatus, String emptyReadinessChecksStatus) {
+        this.modelControllerClientFactory = modelControllerClientFactory;
+        this.managementExecutor = managementExecutor;
         this.emptyLivenessChecksStatus = emptyLivenessChecksStatus;
         this.emptyReadinessChecksStatus = emptyReadinessChecksStatus;
     }
 
     @Override
     public void start(StartContext context) {
         healthReporter = new HealthReporter(emptyLivenessChecksStatus, emptyReadinessChecksStatus);
+
+        modelControllerClient = modelControllerClientFactory.get().createClient(managementExecutor.get());
+
+        if (System.getProperty("__MP_HEALTH_TCK_DISABLE_SERVER_CHECKS") == null) {
```

Comment:
So it basically disable all three checks. Can we remove `_TCK_`part so it does not look like some workaround flag. Then we can also advertise that as feature compatibility flag. If for some reason new behaviour  make problem to someone.

Reply:
that's a good point, I should replace that code by a proper call to MP Config and check the `mp.health.disable-default-procedures` config property instead

Reply:
fixed